src/data/tabular/adult.jl

```diff
@@ -0,0 +1,64 @@
+"""
+    load_uci_adult(n::Union{Nothing, Int}=1000)
+
+Load and preprocesses data from the UCI 'Adult' dataset
+
+# Arguments
+- `n::Union{Nothing, Int}=nothing`: The number of samples to subsample from the dataset.
+
+# Returns
+- `counterfactual_data::CounterfactualData`: A `CounterfactualData` object containing the preprocessed data.
+
+# Example
+data = load_uci_adult(20) # loads and preprocesses 20 samples from the Adult dataset
+"""
+function load_uci_adult(n::Union{Nothing,Int}=1000)
+    # Throw an exception if n < 1:
+    if !isnothing(n) && n < 1
+        throw(ArgumentError("n must be >= 1"))
+    end
```

Comment:
I think an error should also be thrown when n is bigger than the size of the dataset. For example, in the case of German credit data, an error is thrown when n > 1000.